### ISSUE TITLE
Audio bitrate is corrupted

### ISSUE THREAD

**Issue by hakito** [sskodje/ScreenRecorderLib#48]
When recording with the TestApp using default settings + Audio input the resulting video has a time compressed/stretched audio (depending on the player). I inspected the files using MediaInfo:

```
Duration                                 : 3 s 334 ms
Bit rate mode                            : Constant
Bit rate                                 : 48.0 kb/s
Nominal bit rate                         : 96.0 kb/s
Channel(s)                               : 2 channels
```

There should be no nominal bit rate and the Bit rate should be at 96.0 kb/s.

Another recording with debugging enabled produced the following:

```
Bit rate mode                            : Constant
Bit rate                                 : 56.4 kb/s
Nominal bit rate                         : 96.0 kb/s
Channel(s)                               : 2 channels
Channel layout                           : L R
Sampling rate                            : 48.0 kHz
Frame rate                               : 46.875 FPS (1024 SPF)
Compression mode                         : Lossy
Stream size                              : 148 Bytes (0%)
```
And the according debug output:
```
Resampled audio from 16000hz to 48000hz
Wrote frame with start pos 0 ms and with duration 0 ms
Inserted 80 bytes of silence
Wrote audio sample with start pos 0 ms and with duration 0 ms
Received stop event after 734 passes and 58720 frames
Received stop event after 742 passes and 0 frames
Exiting recording task
```

**Comment by hakito** [comment]
My first guess after looking at the implementation is, that because audio is recorded in a separate thread but the sampling is based on the video frame timings the recorded audio is compressed/stretched to the frame duration.

**Comment by sskodje** [comment]
It might be due to the sound being upsampled. What is your audio input device?

**Comment by hakito** [comment]
It's not the resampling. I could reproduce the problem on 3 of 3 machines (notebook and 2 towers). On my current PC there was [no resampling](https://github.com/hakito/ScreenRecorderLib/blob/7999ab76e86c0d03fc886c4cea0f0a042ace17c5/ScreenRecorderLib/loopback_capture.cpp#L156).

**Comment by sskodje** [comment]
That's really odd. I can't reproduce the issue at all. Is the device you are recording audio from a bluetooth or usb device, or are you recording from a regular mic or from the pc output itself?

**Comment by hakito** [comment]
No bluetooth device. USB and audio jack (regular mic). I also disabled to record the output device in the recording settings. For me it's the other way round. Up to now all my recordings on my work PC are corrupted. I had a look at how captura is handling the audio with the media framework. Instead of calculating the start time and duration of a based on actual timings it calculates it using [the buffer size and bitrate](https://github.com/MathewSachin/Captura/blob/3352832fcf0a676a499a4cde9ca067887e0c981f/src/Captura.Windows/MediaFoundation/MfWriter.cs#L286). It's also writing to the sink in a separate thread. So it seems MediaFoundation is thread safe when writing to a separate stream (although I couldn't find it in the docs). I'll try that tomorrow.

**Comment by hakito** [comment]
I think I nailed it. It can also be reproduced recording an output device. I wanted to record mono as this makes sense for a single mic. So I set AudioOptions.Channels to Mono. But the mics I worked with create a stereo signal.

MediaInfo for Mono recording
```
Bit rate                                 : 192 kb/s
Nominal bit rate                         : 96.0 kb/s
Channel(s)                               : 1 channel
```

MediaInfo for Stereo recording
```
Bit rate                                 : 96.0 kb/s
Channel(s)                               : 2 channels
```

However I also [added some logging information about the actual audio sample duration](https://github.com/hakito/ScreenRecorderLib/blob/55d0d5fa967c39e324800e1d075fb47a259dc9b7/ScreenRecorderLib/loopback_capture.cpp#L373) compared to the duration written to the stream. 

```
No resampling nescessary
Probably spurious glitch reported on first packet
Audio duration: 0 total: 0
Wrote frame with start pos 0 ms and with duration 0 ms
Inserted 100 bytes of silence
Wrote audio sample with start pos 0 ms and with duration 0 ms
Audio duration: 200 total: 200
Wrote frame with start pos 0 ms and with duration 194 ms
Wrote audio sample with start pos 0 ms and with duration 194 ms
Audio duration: 100 total: 300
Wrote frame with start pos 194 ms and with duration 104 ms
Wrote audio sample with start pos 194 ms and with duration 104 ms
```

So it's _almost_ in sync. However this can be discussed in a separate issue.

**Comment by hakito** [comment]
BTW. The logging info is not in the PR

**Comment by sskodje** [comment]
Fixed with a51035201a6a5e47d6eff701b8fbeee3685eaff5
